cmake/modules/AddSwift.cmake

```diff
@@ -499,7 +499,13 @@ function(_add_swift_library_single target name)
   endif()
 
   if (SWIFT_COMPILER_VERSION)
-    if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+    if ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "OSX" OR
+        "${SWIFTLIB_SINGLE_SDK}" STREQUAL "IOS" OR
+        "${SWIFTLIB_SINGLE_SDK}" STREQUAL "IOS_SIMULATOR" OR
+        "${SWIFTLIB_SINGLE_SDK}" STREQUAL "TVOS" OR
+        "${SWIFTLIB_SINGLE_SDK}" STREQUAL "TVOS_SIMULATOR" OR
+        "${SWIFTLIB_SINGLE_SDK}" STREQUAL "WATCHOS" OR
+        "${SWIFTLIB_SINGLE_SDK}" STREQUAL "WATCHOS_SIMULTOR")
```

Comment:
This duplication is error-prone.  (We also have another copy in `_add_variant_c_compile_link_flags`.)  Would you mind adding a helper function?